Nitric.Sdk/Document/CollectionRef.cs

```diff
@@ -41,6 +43,22 @@ public DocumentRef<T> Doc(string documentId)
                 this,
                 documentId);
         }
+        public CollectionRef<T> Collection(string name)
+        {
+            if (string.IsNullOrEmpty(name))
+            {
+                throw new ArgumentNullException(name);
+            }
+            if (Util.CollectionDepth(0, this.ToGrpcCollection()) >= Constants.DEPTH_LIMIT)
+            {
+                throw new NotSupportedException("Currently subcollection are only able to be nested " + Constants.DEPTH_LIMIT + "deep");
+            }
+            var parentKey = new Key<T>(this, "");
+            return new CollectionRef<T>(
+                this.documentClient,
+                name,
+                parentKey);
+        }
```

Comment:
In other SDKs the normal key/collection chain is being broken by use of a new `CollectionGroupRef` object, that only contains a `Collection` and `Query` method. This helps prevents users from retrieving broken document references, by making it impossible.